volume/thin_pool_watcher.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package volume
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+// ThinPoolWatcher maintains a cache of device name -> usage stats for a devicemapper thin-pool using thin_ls.
+type ThinPoolWatcher struct {
+	poolName string
+	lock     *sync.RWMutex
+	cache    map[string]uint64
+}
+
+// NewThinPoolWatcher returns a new ThinPoolWatcher for the given devicemapper thin pool name.
+func NewThinPoolWatcher(poolName string) *ThinPoolWatcher {
+	return &ThinPoolWatcher{poolName, &sync.RWMutex{}, make(map[string]uint64)}
+}
+
+func (w *ThinPoolWatcher) Start() {
+	// NOTE: placeholder code, I intend to find a way to reuse kube wait.Forever here
```

Comment:
Yeah, I was going to try to find a way to copy the code, not vendor kube.  I refuse to vendor kube, so I will probably just implement something new.